server/lib/chat/chat-models.ts

```diff
@@ -578,3 +579,105 @@ export async function findChannelByName(
     done()
   }
 }
+
+/**
+ * Returns a list of non-private chat channels that a user is not joined in, optionally filtered by
+ * a `searchStr`.
+ */
+export async function listChannels(
```

Comment:
If this is returning unjoined channels only, it should probably be called `listUnjoinedChannels` or `listJoinableChannels` (probably better considering this filters only visible channels). Alternatively you could make that an option (and probably control it in the API with a query parameter), which would make a lot more sense from a REST perspective